### PR TITLE
[feature] Pagination Changes

### PR DESCRIPTION
**Pull request by lyndsysimon**
…d + made the pagination responsive

Conflicts:
	website/static/js/paginator.js
	website/static/js/tests/paginator.test.js

----

# Screenshot of UI after change

image

### CODE REVIEW

**Comment by samanehsan** [comment]
Does this correspond to a github issue?

**Comment by chennan47** [comment]
The related issue is https://github.com/CenterForOpenScience/osf.io/issues/286, but this needs fix the junit testing.

**Comment by lyndsysimon** [comment]
@ZobairAlijan - could you fix the failing junit tests and push the changes?

**Comment by ZobairAlijan** [comment]
Absolutely.

File: website/static/js/tests/paginator.test.js [hunk 1]
```diff
@@ -47,9 +47,9 @@ describe('Paginator', () => {
     });
 
     it('nextPage', () => {
-        numberOfPages = 5;
+        numberOfPages = 16;
         paginator.configure(function(p){
-            p.numberOfPages(numberOfPages);
+            p.num`erOfPages(numberOfPages);
```
Comment (samanehsan) [inline website/static/js/tests/paginator.test.js]: Is this a typo?

Comment (ZobairAlijan) [inline website/static/js/tests/paginator.test.js]: Hi saman
Yes, I think

**Comment by samanehsan** [comment]
@ZobairAlijan, @lyndsysimon can either of you include screen shots of this UI change? Thanks!

File: website/static/js/tests/paginator.test.js (hunk 1 above)
Comment (samanehsan) [inline website/static/js/tests/paginator.test.js]: Okay, please change that to `numberOfPages`

File: website/static/js/paginator.js [hunk 2]
```diff
@@ -4,8 +4,8 @@
 'use strict';
 var ko = require('knockout');
 var oop = require('js/oop');
-var MAX_PAGES_ON_PAGINATOR = 7;
-var MAX_PAGES_ON_PAGINATOR_SIDE = 5;
+var MAX_PAGES_ON_PAGINATOR = 16;
+var MAX_PAGES_ON_PAGINATOR_SIDE = 16;
```
Comment (samanehsan) [inline website/static/js/paginator.js]: Why is `MAX_PAGES_ON_PAGINATOR` the same as `MAX_PAGES_ON_PAGINATOR_SIDE`? Shouldn't `MAX_PAGES_ON_PAGINATOR_SIDE` be less?

File: website/static/js/paginator.js [hunk 3]
```diff
@@ -62,7 +62,7 @@ var Paginator = oop.defclass({
                     style: 'disabled',
                     text: '...',
                     handler: function() {}
-                });
+                }); // 19 - 15 = 4
```
Comment (samanehsan) [inline website/static/js/paginator.js]: Please remove this comment

File: website/static/js/tests/paginator.test.js [hunk 4]
```diff
@@ -93,8 +94,8 @@ describe('Paginator', () => {
         });
 
         it('more than 7 pages, currentPage less than 4, one ellipse at the end', () => {
```
Comment (samanehsan) [inline website/static/js/tests/paginator.test.js]: Please change this comment to match the test.

File: website/static/js/tests/paginator.test.js [hunk 5]
```diff
@@ -109,8 +110,8 @@ describe('Paginator', () => {
 
         it('more than 7 pages, currentPage more than numbersOfPages - 5, one ellipse at the beginning',
             () => {
```
Comment (samanehsan) [inline website/static/js/tests/paginator.test.js]: Same as above -- please change this comment to reflect the changes you made for the test.

File: website/static/js/tests/paginator.test.js [hunk 6]
```diff
@@ -125,22 +126,22 @@ describe('Paginator', () => {
 
         });
 
-        it('more than 7 pages, currentPage more than 5 and numbersOfPages - 5, two ellipses',
+        it('more than 41 pages, currentPage more than 20 and numbersOfPages - 20, two ellipses',
             () => {
```
Comment (samanehsan) [inline website/static/js/tests/paginator.test.js]: I don't think this comment is quite right -- This says that the number of pages is more than 41, but in the test it is equal to 41. Same with the current page -- the comment says it's more than 20 but it's equal to 20.

**Comment by lyndsysimon** [comment]
@samanehsan - @ZobairAlijan sent me a screenshot, and I've added it to the description.

**Comment by lyndsysimon** [comment]
The pagination breaks on small screens - here's a screenshot at 360x640px, as would be seen on a Samsung Galaxy S4:

image

**Comment by lyndsysimon** [comment]
The bar also wraps at 1024px wide.

image

**Comment by lyndsysimon** [comment]
In short - the behavior of the paginator is currently inconsistent, depending on the resolution of the browser used. To fix this, let's first define exactly how the paginator should respond as the resolution changes, then implement to that standard.

Instead of using Javascript to determine how many buttons are included on the page, consider using [Bootstrap's responsive utility classes](http://getbootstrap.com/css/#responsive-utilities) to control how many are visible. We should at least be testing that things are displayed correctly at all four Bootstrap-defined sizes: `xs`, `sm`, `md`, and `lg`.

**Comment by lyndsysimon** [comment]
After speaking with @ZobairAlijan yesterday, I'm closing this PR. This feature will still likely be implemented, but the direction has changed somewhat in *how*.

Please submit a new PR when ready for review.

**Pull request closed without merging**